php-classes/Slate/CBL/StudentCompetency.php

```diff
@@ -53,16 +63,334 @@ class StudentCompetency extends \ActiveRecord
         ]
     ];
 
-    public static function isCurrentLevelComplete($Student, $Competency)
+    public static $dynamicFields = [
+        'demonstrationsLogged' => [
+            'getter' => 'getDemonstrationsLogged'
+        ],
+        'demonstrationsComplete' => [
+            'getter' => 'getDemonstrationsComplete'
+        ],
+        'demonstrationsAverage' => [
+            'getter' => 'getDemonstrationsAverage'
+        ],
+        'demonstrationData' => [
+            'getter' => 'getDemonstrationData'
+        ],
+        'effectiveDemonstrationsData' => [
+            'getter' => 'getEffectiveDemonstrationsData'
+        ]
+
+    ];
+
+    public function calculateStartingRating($autoSave = true)
     {
-        $completion = $Competency->getCompletionForStudent($Student);
+        $validSkills = DB::allRecords(
```

Comment:
This will return null values where there is no first rating, so after this you just want to do:

```php
$ratingsCount = count(array_filter($ratings));
if ($ratingsCount != count($ratings) {
    return null;
}

return array_sum($ratings) / $ratingsCount;;
```